File: jp-minimal-ios/JPMinimal/Visual/ProductEmoji.swift
import Foundation

struct EmojiVisual: Hashable {
    let emoji: String
    let scale: CGFloat
    let label: String
}

enum ProductEmoji {
    private static let fallback = EmojiVisual(emoji: "🛍️", scale: 0.9, label: "Peça")

    private static let map: [String: EmojiVisual] = [
        "tee": .init(emoji: "👕", scale: 1, label: "Camiseta"),
        "tee-long": .init(emoji: "👕", scale: 1.06, label: "Camiseta alongada"),
        "shirt": .init(emoji: "👔", scale: 1, label: "Camisa"),
        "shirt-camp": .init(emoji: "👕", scale: 0.96, label: "Camisa camp"),
        "pants": .init(emoji: "👖", scale: 1.08, label: "Calça"),
        "pants-wide": .init(emoji: "👖", scale: 1.14, label: "Calça wide"),
        "pants-tailored": .init(emoji: "👖", scale: 1.04, label: "Calça de alfaiataria"),
        "shorts": .init(emoji: "🩳", scale: 0.88, label: "Bermuda"),
        "shorts-chino": .init(emoji: "🩳", scale: 0.9, label: "Bermuda chino"),
        "sweater": .init(emoji: "🧶", scale: 0.92, label: "Suéter"),
        "cardigan": .init(emoji: "🧥", scale: 1.08, label: "Cardigan"),
        "turtleneck": .init(emoji: "🧣", scale: 0.86, label: "Gola alta"),
        "blazer": .init(emoji: "🤵", scale: 1.12, label: "Blazer"),
        "jacket": .init(emoji: "🧥", scale: 1.16, label: "Jaqueta"),
        "letterman": .init(emoji: "🧥", scale: 1.12, label: "Jaqueta college"),
        "trucker": .init(emoji: "🧥", scale: 1.08, label: "Jaqueta trucker"),
        "coat": .init(emoji: "🧥", scale: 1.22, label: "Casaco"),
        "puffer": .init(emoji: "🧥", scale: 1.18, label: "Puffer"),
        "parka": .init(emoji: "🧥", scale: 1.2, label: "Parka"),
        "dress": .init(emoji: "👗", scale: 1.2, label: "Vestido"),
        "slip": .init(emoji: "👗", scale: 1.14, label: "Vestido slip"),
        "skirt": .init(emoji: "👗", scale: 0.92, label: "Saia"),
        "skirt-full": .init(emoji: "👗", scale: 1.02, label: "Saia evasê"),
        "sneaker": .init(emoji: "👟", scale: 0.86, label: "Tênis"),
        "loafer": .init(emoji: "👞", scale: 0.84, label: "Sapato"),
        "bag": .init(emoji: "👜", scale: 0.78, label: "Bolsa"),
        "belt": .init(emoji: "🪢", scale: 0.64, label: "Cinto"),
        "cap": .init(emoji: "🧢", scale: 0.7, label: "Boné"),
        "scarf": .init(emoji: "🧣", scale: 0.76, label: "Cachecol"),
        "wallet": .init(emoji: "👛", scale: 0.62, label: "Carteira"),
        "watch": .init(emoji: "⌚", scale: 0.58, label: "Relógio"),
    ]

    static func visual(for motif: String) -> EmojiVisual {
        map[motif] ?? fallback
    }
}

File: jp-minimal-android/app/src/main/java/br/com/jpminimal/visual/ProductEmoji.kt
package br.com.jpminimal.visual

data class EmojiVisual(
    val emoji: String,
    val scale: Float,
    val label: String,
)

private val FALLBACK = EmojiVisual("🛍️", 0.9f, "Peça")

private val MAP = mapOf(
    "tee" to EmojiVisual("👕", 1f, "Camiseta"),
    "tee-long" to EmojiVisual("👕", 1.06f, "Camiseta alongada"),
    "shirt" to EmojiVisual("👔", 1f, "Camisa"),
    "shirt-camp" to EmojiVisual("👕", 0.96f, "Camisa camp"),
    "pants" to EmojiVisual("👖", 1.08f, "Calça"),
    "pants-wide" to EmojiVisual("👖", 1.14f, "Calça wide"),
    "pants-tailored" to EmojiVisual("👖", 1.04f, "Calça de alfaiataria"),
    "shorts" to EmojiVisual("🩳", 0.88f, "Bermuda"),
    "shorts-chino" to EmojiVisual("🩳", 0.9f, "Bermuda chino"),
    "sweater" to EmojiVisual("🧶", 0.92f, "Suéter"),
    "cardigan" to EmojiVisual("🧥", 1.08f, "Cardigan"),
    "turtleneck" to EmojiVisual("🧣", 0.86f, "Gola alta"),
    "blazer" to EmojiVisual("🤵", 1.12f, "Blazer"),
    "jacket" to EmojiVisual("🧥", 1.16f, "Jaqueta"),
    "letterman" to EmojiVisual("🧥", 1.12f, "Jaqueta college"),
    "trucker" to EmojiVisual("🧥", 1.08f, "Jaqueta trucker"),
    "coat" to EmojiVisual("🧥", 1.22f, "Casaco"),
    "puffer" to EmojiVisual("🧥", 1.18f, "Puffer"),
    "parka" to EmojiVisual("🧥", 1.2f, "Parka"),
    "dress" to EmojiVisual("👗", 1.2f, "Vestido"),
    "slip" to EmojiVisual("👗", 1.14f, "Vestido slip"),
    "skirt" to EmojiVisual("👗", 0.92f, "Saia"),
    "skirt-full" to EmojiVisual("👗", 1.02f, "Saia evasê"),
    "sneaker" to EmojiVisual("👟", 0.86f, "Tênis"),
    "loafer" to EmojiVisual("👞", 0.84f, "Sapato"),
    "bag" to EmojiVisual("👜", 0.78f, "Bolsa"),
    "belt" to EmojiVisual("🪢", 0.64f, "Cinto"),
    "cap" to EmojiVisual("🧢", 0.7f, "Boné"),
    "scarf" to EmojiVisual("🧣", 0.76f, "Cachecol"),
    "wallet" to EmojiVisual("👛", 0.62f, "Carteira"),
    "watch" to EmojiVisual("⌚", 0.58f, "Relógio"),
)

fun getProductEmoji(motif: String): EmojiVisual = MAP[motif] ?: FALLBACK
